scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -133,10 +133,12 @@ def _verify_ztest_suite_name(self, harness_state, detected_suite_names, handler_
             return
         if not detected_suite_names:
             self._missing_suite_name(expected_suite_names, handler_time)
-        for detected_suite_name in detected_suite_names:
-            if detected_suite_name not in expected_suite_names:
+            return
+        # compare the expect and detect from end one by one without order
+        _d_suite = detected_suite_names[-len(expected_suite_names):]
+        if set(_d_suite) != set(expected_suite_names):
+            if not set(_d_suite).issubset(set(expected_suite_names)):
```

Comment:
Are the erroneous suite names always in the front, or is it just highly likely?

Additionally, in the example you've provided, suite names repeat. This information will be lost when using `set()`. Will this have any negative repercussions? Or should we use `all([suite_name in expected_suite_names for suite_name in _d_suite])`- instead?

Reply:
> Are the erroneous suite names always in the front, or is it just highly likely?
yes. as those are coming from last case.





> This information will be lost when using `set()`. Will this have any negative repercussions? Or should we use `all([suite_name in expected_suite_names for suite_name in _d_suite])`- instead?

with the first commit in this PR, the repeat suite name is removed if they are intrduced by repeated scan. the thing code comes to this condition is there are cases in cmsis_dsp test, which you expected multiply test suites but only one detected, due to build time flag, one one type of cases are executed.

Reply:
when the detected names have more items than expected and the last items are the same, doesn't the new code lead to a false positive for the expected test ?

Reply:
@golowanow , the has been filter out by  `if set(_d_suite) != set(expected_suite_names):`